python/lib/physiological.py

```diff
@@ -549,6 +577,162 @@ def insert_event_file(self, event_data, event_file, physiological_file_id,
             physiological_file_id, 'event_file_blake2b_hash', blake2
         )
 
+    def insert_annotation_metadata(self, annotation_metadata, annotation_metadata_file, physiological_file_id,
+                          blake2):
+        """
+        Inserts the annotation metadata information read from the file *annotations.json
+        into the physiological_annotation_file, physiological_annotation_parameter
+        and physiological_annotation_label tables, linking it to the physiological file ID
+        already inserted in physiological_file.
+
+        :param annotation_metadata      : list with dictionaries of annotations
+                                          metadata to insert into the database
+         :type annotation_metadata      : list
+        :param annotation_metadata_file : name of the annotation metadata file
+         :type annotation_file          : str
+        :param physiological_file_id    : PhysiologicalFileID to link the annnotation info to
+         :type physiological_file_id    : int
+        :param blake2                   : blake2b hash of the annotation file
+         :type blake2                   : str
+        """
+
+        optional_fields = (
+            'Sources',  'Author', 'LabelDescription'
+        )
+        for field in optional_fields:
+            if field not in annotation_metadata.keys():
+                annotation_metadata[field] = None
+
+        annotationFileID = self.db.insert(
+            table_name   = 'physiological_annotation_file',
+            column_names = ('PhysiologicalFileID', 'FileType', 'FilePath'),
+            values       = (physiological_file_id, 'json',  annotation_metadata_file),
+            get_last_id  = True
+        )
+
+        self.db.insert(
+            table_name   = 'physiological_annotation_parameter',
+            column_names = ('AnnotationFileID', 'Sources',  'Author'),
+            values       = (annotationFileID, annotation_metadata['Sources'], annotation_metadata['Author'])
+        )
+
+        labels = []
+        for label, description in annotation_metadata['LabelDescription'].items():
+            descCount = self.db.pselect(
+                query="SELECT COUNT(*) FROM physiological_annotation_label WHERE LabelName = %s",
+                args=(label,)
+            )
+
+            if descCount == 0:
+                labels.insert((label, description))
+
+        self.db.insert(
+            table_name   = 'physiological_annotation_label',
+            column_names = ('LabelName', 'LabelDescription'),
+            values       = labels
+        )
+
+        # insert blake2b hash of annotation file into physiological_parameter_file
+        self.insert_physio_parameter_file(
```

Comment:
this particular case has not yet been refactored and can be tacked on later on. I just think that for new queries, we should try to put all the database queries in the /database_lib/ folder, one file per table queried. One less thing to refactor after ;)